plane/src/proxy/rewriter.rs

```diff
@@ -16,6 +20,18 @@ const PATH_PREFIX_HEADER: &str = "x-verified-path";
 const X_FORWARDED_FOR_HEADER: &str = "x-forwarded-for";
 const X_FORWARDED_PROTO_HEADER: &str = "x-forwarded-proto";
 
+#[derive(Debug, thiserror::Error)]
+pub enum RequestRewriterError {
+    #[error("Invalid `host` header")]
+    InvalidHostHeader,
+}
+
+impl From<hyper::header::ToStrError> for RequestRewriterError {
+    fn from(_: hyper::header::ToStrError) -> Self {
+        RequestRewriterError::InvalidHostHeader
+    }
+}
```

Comment:
Yes, this is a good and idiomatic use of `From`!

You may have noticed that `?` will do a conversion when there is a `From` implementation, so something like:

```rust
fn blah() -> Result<(), ToStrError> {
    let c: Result<(), RequestRewriterError> = blahblah();
    c?;
    Ok(())
}
```

will work thanks to this implementation.